tokio-uds/src/listener.rs

```diff
@@ -43,15 +45,21 @@ impl UnixListener {
     }
 
     /// Test whether this socket is ready to be read or not.
-    pub fn poll_read_ready(&self, ready: Ready) -> Poll<Ready, io::Error> {
-        self.io.poll_read_ready(ready)
+    pub fn poll_read_ready(&self, cx: &mut Context<'_>, ready: Ready) -> Poll<io::Result<Ready>> {
+        self.io.poll_read_ready(cx, ready)
     }
 
     /// Returns the value of the `SO_ERROR` option.
     pub fn take_error(&self) -> io::Result<Option<io::Error>> {
         self.io.get_ref().take_error()
     }
 
+    /// Returns a future that attempts to accept a connection and creates a new
+    /// connected `UnixStream` if successful.
+    pub fn accept(&mut self) -> Accept<'_> {
```

Comment:
could prob just make this a `async fn` I believe that is what we do in `tokio-tcp` now. Internally this can just call out to `async-util`'s poll_fn. There should be usage examples within tokio already.